lib/node/nodes/georeference-street-address.js

```diff
@@ -49,42 +53,58 @@ GeoreferenceStreetAddress.prototype.getGeocoderParams = function () {
     return geocoderParams;
 };
 
-var COLUMNS_TEMPLATE_REGEX = /\{\{([^\}]*)\}\}/g;
+var COLUMNS_TEMPLATE_REGEX = /\{\{([^\}]*)\}\}|([^{]+)/g;
 GeoreferenceStreetAddress.prototype.getStreetAddressColumnParam = function() {
     if (this.street_address_column) {
         return this.getFunctionParam('street_address');
     }
 
-    var columns = [];
-    var result = this.street_address_template.replace(COLUMNS_TEMPLATE_REGEX, function(s, column) {
-        columns.push(column.trim());
-        return '%s';
-    });
+    // Let's split columns and literals to escape literals and concat everything. Example:
+    // "{{c1}} {{c2}}, {{c3}}, Spain" --> ["c1", "c2", "', '", "c3", "', Spain'"]
+    var columnsAndLiterals = [];
+    var addressMatch;
+    while ((addressMatch = COLUMNS_TEMPLATE_REGEX.exec(this.street_address_template)) !== null) {
+        var column = addressMatch[1];
+        if(column) {
+            columnsAndLiterals.push(column.trim());
+        } else {
+            columnsAndLiterals.push(`'${addressMatch[0]}'`);
+        }
+    }
 
-    if (columns.length === 0) {
-        return '$tpl$' + this.street_address_template + '$tpl$';
+    if (columnsAndLiterals.length === 0) {
+        return `$tpl$'${this.street_address_template}'$tpl$`;
```

Comment:
Just wondering. Why did we have to abandon the `format` approach? I guess we cannot pass that to the remote server to be lazily evaluated but we can do it with a string. 🤔

Reply:
The reason was the lazy evaluation. Previously, extracting the columns was enough. Now, to enable lazy evaluation, literals must be escaped.

I don't know what you mean by "doing it with a string", could you elaborate, please? TBH, this was one of the first changes, maybe it's not needed. Take a look at dataservices side as well. This kind of design decisions are the ones that made me ask for your opinion on this :-)

Reply:
Sorry about my wording. By "doing it with a string", I meant something similar to what you say with "literals must be escaped". I tend to use `format` over string concatenation, but I guess it's not an option here.

Reply:
Yep, I think that it's not possible (or makes things more complicated, I'm not sure).

Reply:
Let's :shipit: like this.